src/librustc/hir/map/mod.rs

```diff
@@ -713,14 +713,17 @@ impl<'hir> Map<'hir> {
                 Node::Item(_) | Node::ForeignItem(_) | Node::TraitItem(_) | Node::ImplItem(_) => {
                     break;
                 }
-                Node::Expr(expr) => match expr.kind {
-                    ExprKind::Match(_, _, _) => return Some(expr),
-                    _ => {}
-                },
-                Node::Stmt(stmt) => match stmt.kind {
-                    StmtKind::Local(_) => break,
-                    _ => {}
-                },
+                Node::Expr(expr) => {
```

Comment:
drive-by comment: Maybe this'd be better if we matched directly on
```rust
Node::Expr(expr @ ExprKind::Match(...)) => return Some(expr),
```
and similarly below?

Reply:
Hm, this won't compile
```
error[E0308]: mismatched types
   --> src/librustc/hir/map/mod.rs:716:35
    |
712 |             match node {
    |                   ---- this expression has type `rustc_hir::hir::Node<'_>`
...
716 |                 Node::Expr(expr @ ExprKind::Match(..)) => {return Some(expr)},
    |                                   ^^^^^^^^^^^^^^^^^^^ expected struct `rustc_hir::hir::Expr`, found enum `rustc_hir::hir::ExprKind`
```
I'm not very familiar with this kind of syntax myself :(

Reply:
Oh, whoops, that’s matching on the field `kind`. Sorry for the trouble 🙇